packages/api/src/auth/decoders/magicLink.ts

```diff
@@ -1,13 +1,28 @@
-export const magicLink = (token: string) => {
-  if (!process.env.MAGICLINK_PUBLIC) {
-    if (process.env.NODE_ENV === 'development') {
-      console.warn(
-        'MAGICLINK_PUBLIC env var is not set. Be certain to set this value in Production.'
-      )
-    } else {
-      console.error('Magic Auth configuration error.')
-      throw new Error('Magic Auth configuration error.')
-    }
+// Relevant type excerpt from Magiclink's magic-sdk/admin:
+// export interface Claim {
+//   iat: number; // Issued At Timestamp
+//   ext: number; // Expiration Timestamp
+//   iss: string; // Issuer of DID Token
+//   sub: string; // Subject
+//   aud: string; // Audience
+//   nbf: number; // Not Before Timestamp
+//   tid: string; // DID Token ID
+//   add: string; // Encrypted signature of arbitrary data
+// }
+// export type ParsedDIDToken = [string, Claim];
+//
+// This function validates the token, throw an error if validations fails
+// otherwise it will return the decoded token in the form of a ParsedDIDToken
+export const magicLink = async (token: string) => {
+  const { MAGIC_SECRET_API_KEY } = process.env
+  if (!MAGIC_SECRET_API_KEY) {
+    throw new Error('`MAGIC_SECRET_API_KEY` environment variable not set.')
   }
-  return token
+  const { Magic } = require('@magic-sdk/admin')
+
+  const magicAdmin = new Magic(MAGIC_SECRET_API_KEY)
+
+  await magicAdmin.token.validate(token)
+
+  return magicAdmin.token.decode(token)
```

Comment:
This will return `[proof, claim]`.

```javascript
  console.log(proof); // => string

  console.log(claim);
  /*
    => {
      iat: number,
      ext: number,
      iss: string,
      sub: string,
      aud: string,
      nbf: number,
      tid: string
    }
   */
```

I assume the decoder's work is to decode the token and return what's inside. Right? If that's the case. This PR LGTM 👍

Reply:
The return value of this function ends up being the decoded argument in the end-user's application's getCurrentUser() function.  Here's the relevant docs for the old version where the decoded argument is just discarded since there was no decoder previously: https://redwoodjs.com/docs/authentication#setup-13 

The only thing holding me back from moving this out of draft and into 'ready to merge' is that I haven't tested it myself.